packages/navigator/src/browser/navigator-model.ts

```diff
@@ -112,4 +165,41 @@ export class FileNavigatorModel extends FileTreeModel {
         return Iterators.cycle(filteredNodes, node);
     }
 
+    protected collectAffectedNodes(change: FileChange, accept: (node: CompositeTreeNode) => void): void {
+        if (this.isFileContentChanged(change)) {
+            return;
+        }
+        this.getNodesByUri(change.uri.parent.toString()).forEach(parentNode => {
+            if (DirNode.is(parentNode) && parentNode.expanded) {
+                accept(parentNode);
+            }
+        });
+    }
+
+    protected isFileContentChanged(change: FileChange): boolean {
+        if (this.roots.length === 0) {
+            return false;
+        }
+        const nodeId = this.getNavigatorFileTreeNodeId(this.roots[0], change.uri.toString());
+        const node = this.getNode(nodeId);
+        return change.type === FileChangeType.UPDATED && FileNode.is(node);
+    }
+
+    protected getNodesByUri(nodeUri: string): TreeNode[] {
+        const nodes: TreeNode[] = [];
+        for (const node of this.roots.map(root => this.getNavigatorFileTreeNodeId(root, nodeUri))
+            .map(parentNodeId => this.getNode(parentNodeId))) {
+            if (node) {
+                nodes.push(node);
+            }
+        }
+        return nodes;
+    }
+
+    private getNavigatorFileTreeNodeId(rootPath: string, nodeUri: string): string {
+        if (!this.hasMultipleRoots && rootPath === nodeUri) {
+            return nodeUri;
+        }
+        return `${new URI(rootPath).path.toString()}///${nodeUri}`;
```

Comment:
This tree node ID logic should be moved to a factory method or function. It will be easier to ~fix any future bugs~ maintain the code in one place.